lib/install.js

```diff
@@ -0,0 +1,161 @@
+const _ = require('lodash')
+const request = require('request')
+const Promise = require('bluebird')
+const figures = require('figures')
+const exec = require('child_process').exec
+
+const prefix = 'npm-addon-'
+const resolve = require('url').resolve
+
+exports.command = 'install <addon>'
+exports.describe = 'installs a new OAuth 2.0 client from a package'
+exports.builder = function (yargs) {
+  return yargs
+    .option('registry', {
+      alias: 'r',
+      describe: 'npm registry URL',
+      default: 'https://registry.npmjs.org'
+    })
+    .option('port', {
+      alias: 'p',
+      describe: 'port server is running on',
+      default: 8084
+    })
+}
+
+exports.handler = function (argv) {
+  var addon = prefix + argv.addon
+  var meta = null
+  var client = null
+
+  exists(addon, argv)
+    .then(function () {
+      console.info(figures.tick + ' found addon "' + argv.addon + '"')
+      return install(addon, argv)
+    })
+    .then(function () {
+      console.info(figures.tick + ' installed addon "' + argv.addon + '"')
+      meta = require(addon)
+      return createClient(meta, argv)
+    })
+    .then(function (_client) {
+      client = _client
+      console.info(figures.tick + ' generated client for "' + argv.addon + '"')
+      return createToken(client, argv)
+    })
+    .then(function (token) {
+      console.info(figures.tick + ' generated access token for "' + argv.addon + '"')
+      return sendToken(client, token, argv)
+    })
+    .then(function () {
+      console.info('\\o/ addon "' + argv.addon + '" successfully installed')
+    })
+    .catch(function (err) {
+      console.error(err.message)
+    })
+}
+
+function exists (addon, argv) {
+  const deferred = Promise.defer()
+
+  request.get(resolve(argv.registry, addon), function (err, res) {
```

Comment:
In cases of read-thru-cache for npmE, couldn't this return a 404 even though `npm install` might work (by adding it to the whitelist)?